### PR TITLE
Use vue recommended ESLint rules

### LINKED ISSUE #71: Determine which Vue ESLint rules to adopt
## Overview
Vue CLI projects out of the box use `plugin:vue/essential` which is considered Priority A. There are 4 choices:

### Base Rules 
**Goal**: Enabling Correct ESLint Parsing
- `plugin:vue/base`
### Priority A: Essential
**Goal**: Error Prevention
- `plugin:vue/essential`
### Priority B: Strongly Recommended 
**Goal**: Improving Readability
- `plugin:vue/strongly-recommended`
### Priority C: Recommended 
**Goal**: Minimizing Arbitrary Choices and Cognitive Overhead
- `plugin:vue/recommended`

More can be read about the rules at https://eslint.vuejs.org/rules/

### PR DESCRIPTION
**Pull request by derick-montague**
- Changed the ESLint rules to use  Priority 4: Recommended and updated
all the files that needed to be updated.
- Changed vue/component-name-in-tempate-casing to use kebab-case
naming since Bootstrap vue plugin uses that naming convention.
ESLint Strongly recommended defaults to PascalCase, but kebab-case is
also listed as acceptable.

Resolves: #71

Signed-off-by: Derick Montague <derick.montague@ibm.com>

### CODE REVIEW

File: src/components/Global/PageTitle.vue [hunk 1]
```diff
@@ -8,7 +8,12 @@
 <script>
 export default {
   name: 'PageTitle',
-  props: ['description'],
+  props: {
```
Comment (yoshiemuranaka) [inline src/components/Global/PageTitle.vue:11]: How does everyone feel about this? Adding type to props?

File: src/components/AppNavigation/AppNavigation.vue [hunk 2]
```diff
@@ -45,7 +45,7 @@
         <icon-access-control />Access Control
         <icon-expand class="icon-expand" />
       </b-button>
-      <b-collapse tag="ul" id="access-control-menu" class="nav-item__nav">
+      <b-collapse id="access-control-menu" tag="ul" class="nav-item__nav">
```
Comment (yoshiemuranaka) [inline src/components/AppNavigation/AppNavigation.vue:48]: Will we get build errors when attributes aren't in the right order?

**Review by yoshiemuranaka — COMMENTED** [review]
Will we get build errors or warnings for these formatting rules? Already finding some of the rules like requiring semicolon cumbersome (ie semicolon after console log for quick debugging). I'm for auto fixing these rules but would be nice if we can change some to warn instead of error so it is less disruptive to the workflow

File: src/components/Global/PageTitle.vue (hunk 1 above)
Comment (derick-montague) [reply src/components/Global/PageTitle.vue:11]: I personally like it. It's nice that the errors can be provided to the user if they don't pass an required prop or the right type.

File: src/components/AppNavigation/AppNavigation.vue (hunk 2 above)
Comment (derick-montague) [reply src/components/AppNavigation/AppNavigation.vue:48]: I'm not sure if these are warnings or errors. Running `npm run lint` auto fixes them. I expect they will be auto fixed if you have your editor set to auto fix eslint on save.

**Comment by derick-montague** [comment]
> Will we get build errors or warnings for these formatting rules? Already finding some of the rules like requiring semicolon cumbersome (ie semicolon after console log for quick debugging). I'm for auto fixing these rules but would be nice if we can change some to warn instead of error so it is less disruptive to the workflow.

Semi-colons should be an error if we are expecting them. I don't know that we can turn it on or off for specifics. We could update to only error on build. you can see how that works in the eslintrc for console.log and debugger. 

The other solution is to have the eslint auto fix on save, which will add the semi-colons for you.

**Review by yoshiemuranaka — APPROVED** [review]
It looks like you can add overrides to the config to change errors to warn, but I'm ok with merging this in and working with it. Maybe we can revisit overrides if anyone else finds the errors too strong/disruptive

**Review by dixsie — APPROVED** [review]

**Comment by yoshiemuranaka** [comment]
There's also a lintOnSave config option that can be changed to 'warning'

> When set to true or 'warning', eslint-loader will emit lint errors as warnings. By default, warnings are only logged to the terminal and does not fail the compilation, so this is a good default for development.

https://cli.vuejs.org/config/#lintonsave

**Comment by derick-montague** [comment]
Migrated to webui-vue using https://gerrit.openbmc-project.xyz/c/openbmc/webui-vue/+/28763

**Pull request closed without merging**